stats.c

```diff
@@ -688,6 +716,142 @@ void update_checksum(bam1_t *bam_line, stats_t *stats)
     stats->checksum.quals += crc32(0L, qual, (seq_len+1)/2);
 }
 
+static int collect_barcode_stats(bam1_t* bam_line, stats_t* stats, const char *barcode_tag, const char *qual_tag) {
+    // Barcode statistics
+    uint32_t i;
+    uint8_t* bc = bam_aux_get(bam_line, barcode_tag);
+    if (bc) {
+        char* barcode = bam_aux2Z(bc);
+        if (barcode) {
+            uint32_t barcode_len = strlen(barcode), nbases;
+            int *maxqual;
+            acgtno_count_t *acgtno;
+            uint64_t *quals;
+            int32_t *separator;
+
+            if (!strncmp(barcode_tag, "BC", 2)) {
```

Comment:
There is a lot of duplicated code for BC, CR, OX and RX.  I'm wondering if we need an array keyed on the ID.  Even at a 2d array of [128][128] it's still not a huge amount of memory (assuming we likewise malloc the actual data arrays the first time they're observed).  This would permit it to be adjustable and to work with whatever local tags people have created.  Example data with https://github.com/grenaud/deML for example uses XI/YI and XJ/YJ.